python/lsst/ts/mtdomegui/model.py

```diff
@@ -62,11 +125,605 @@ def __init__(
 
         # TODO: Put this into the ts_config_mttcs in DM-48109.
         self.connection_information = {
-            "host": "localhost",
-            "port": 1000,
-            "timeout_connection": 10.0,
+            "host": host,
+            "port": port,
         }
 
         self.duration_refresh = duration_refresh
 
         self.reporter = Reporter(self.log)
+
+        self.mtdome_com: MTDomeCom | None = None
+
+        self.periodic_tasks: list[asyncio.Task] = list()
+
+    async def connect(self) -> None:
+        """Connect to the controller.
+
+        Start the mock controller, if simulating.
+        """
+
+        config = type(
+            "Obj",
+            (object,),
+            {key: value for key, value in self.connection_information.items()},
+        )()
+        self.mtdome_com = MTDomeCom(
+            self.log, config, simulation_mode=self._is_simulation_mode
+        )
+
+        try:
+            await self.mtdome_com.connect()
+        except Exception as error:
+            raise RuntimeError(f"Connection to the controller failed: {error!r}.")
+
+        self.reporter.report_state_azimuth_axis(MTDome.EnabledState.ENABLED)
+        self.reporter.report_fault_code_azimuth_axis("")
+
+        self.reporter.report_state_elevation_axis(MTDome.EnabledState.ENABLED)
+        self.reporter.report_fault_code_elevation_axis("")
+
+        self.reporter.report_state_aperture_shutter(MTDome.EnabledState.ENABLED)
+        self.reporter.report_fault_code_aperture_shutter("")
+
+        self.reporter.report_state_brake_engaged(0)
+        self.reporter.report_state_locking_pins_engaged(0)
+
+        self.reporter.report_state_power_mode(self.mtdome_com.power_management_mode)
+
+        # Start polling for the status of the lower level components
+        # periodically.
+        await self.start_periodic_tasks()
+
+        self.log.info("connected")
+
+    async def start_periodic_tasks(self) -> None:
+        """Start all periodic tasks."""
+
+        await self.cancel_periodic_tasks()
+
+        for method, interval in self.all_methods_and_intervals.items():
+            funcion = getattr(self, method)
+            self.periodic_tasks.append(
+                asyncio.create_task(self.one_periodic_task(funcion, interval))
+            )
+
+        # Workaround the mypy check
+        assert self.mtdome_com is not None
+
+        self.periodic_tasks.append(
+            asyncio.create_task(
+                self.one_periodic_task(
+                    self.mtdome_com.check_all_commands_have_replies,
+                    COMMANDS_REPLIED_PERIOD,
+                )
+            )
+        )
+
+        self.periodic_tasks.append(
+            asyncio.create_task(
+                self.one_periodic_task(
+                    self.mtdome_com.process_command_queue,
+                    _COMMAND_QUEUE_PERIOD,
+                )
+            )
+        )
+
+    async def cancel_periodic_tasks(self) -> None:
```

Comment:
This is an exact copy of the method in ts_mtdome. We should extract the one from ts_mtdome and put it in ts_mtdomecom.